### ISSUE TITLE
ListView onselectionchanging or onselectionchanged fails to call declared function

### ISSUE THREAD

**Issue by omnoms** [winjs/winjs#184]
Scenario;

User choses to add an item, clicks a button and gets navigated to a selection view (this listview among two others), the user clicks an item and a selection is made, an appbar is shown with options and the user clicks to use the item selected and is navigated away to the next page. However, if the user realizes he/she used wrong item, they can now navigate back and select a new one. When the user now clicks a new item in the list, the app crashes with an exception.

So basically the navigation is this;

Root (button-click) -> Item Page (item-click + button-click)-> Edit Page (back)<- Item Page (item-click)

``` javascript
Exception is about to be caught by JavaScript library code at line 669, column 61 in ms-appx://microsoft.phone.winjs.2.1/js/base.js
0x800a138a - JavaScript runtime error: Function expected
File: base.js, Line: 669, Column: 61
```

The markup

``` HTML
            <div id="wlActivities"
                 class="win-selectionstylefilled wlList"
                 data-win-control="WinJS.UI.ListView"
                 data-win-options="{
                    itemTemplate: wlListTemplate,
                    groupHeaderTemplate: wlHeaderTemplate,
                    onselectionchanging: worklistView.selectingItem,
                    onselectionchanged: worklistView.selectedItem,
                    tapBehavior: 'toggleSelect',
                    selectionMode: 'multi',
                    layout: { type: worklistView.MyLayout, groupHeaderPosition: 'top' }
            }"></div>
```

``` javascript
        selectingItem: WinJS.Utilities.markSupportedForProcessing(function (ev) {
            if(ev.detail.newSelection.count() > 1) {
                ev.preventDefault();
            } 
        }),
        selectedItem: WinJS.Utilities.markSupportedForProcessing(function(ev) {
            var el = document.querySelector("#" + ev.target.id).winControl;
            var appbar = document.querySelector("#appbar").winControl;
            var useBtn = document.getElementById("wl_useItem");
            var editBtn = document.getElementById("wl_editItem");
            if (el) {
                if (el.selection.count() === 1) {
                    var item = el.selection.getItems();
                    console.log(item._value[0].data);
                    worklistView.selectedItem = item._value[0].data;
                    var nav = "";
                    switch (ev.target.id) {
                        case 'wlActivities':
                            nav = "Activity";
                            break;
                        case 'wlCases':
                            nav = "Case"
                            break;
                        default:
                    }
                    worklistView.selectedItem.FavouriteType = nav;
                    //show command bar
                    if (appbar) {
                        appbar.disabled = false;
                        appbar.showOnlyCommands(['wl_useItem', 'wl_editItem']);
                        useBtn.addEventListener("click", worklistView.itemClicked, false);
                        editBtn.addEventListener("click", worklistView.editClicked, false);

                    }
                } else {
                    // hide command bar
                    if (appbar) {
                        useBtn.removeEventListener("click", worklistView.itemClicked, false);
                        editBtn.removeEventListener("click", worklistView.editClicked, false);
                        appbar.disabled = true;
                    }
                }
            }
        }),
        editClicked:  WinJS.Utilities.markSupportedForProcessing(function() {

        }),
        MyLayout: WinJS.Class.define(function (options) {
                this._site = null;
                this._surface = null;
            },
            {
                // This sets up any state and CSS layout on the surface of the custom layout
                initialize: function (site) {
                    this._site = site;
                    this._surface = this._site.surface;

                    // Add a CSS class to control the surface level layout
                    WinJS.Utilities.addClass(this._surface, "myLayout");

                    return WinJS.UI.Orientation.vertical;
                },
                // Reset the layout to its initial state
                uninitialize: function () {
                    WinJS.Utilities.removeClass(this._surface, "myLayout");
                    this._site = null;
                    this._surface = null;
                },
            })   
```

The crash happens before the appbar is visible. So I suspect it's onselectionchanging that is bugging.
